tests/test_optimizer_random_dag.py

```diff
@@ -116,13 +130,13 @@ def compare_optimization_results(dag: sky.Dag, minimize_cost: bool):
                                                       dag.tasks, optimizer_plan)
 
     min_objective = find_min_objective(copy_dag, minimize_cost)
-    assert abs(objective - min_objective) < 1e-3
+    assert abs(objective - min_objective) < 5e-2
 
 
 def test_optimizer(enable_all_clouds):
+    for seed in range(3):
+        dag = generate_random_dag(num_tasks=5, seed=seed)
+        sky.Optimizer._add_dummy_source_sink_nodes(dag)
 
-    dag = generate_random_dag(num_tasks=5, seed=0)
-    sky.Optimizer._add_dummy_source_sink_nodes(dag)
-
-    compare_optimization_results(dag, minimize_cost=True)
-    compare_optimization_results(dag, minimize_cost=False)
+        compare_optimization_results(dag, minimize_cost=True)
+        compare_optimization_results(dag, minimize_cost=False)
```

Comment:
```suggestion
```

Maybe we could disable optimize for time since we use a time estimator that only returns a constant number. Therefore, the best plan is the first encountered one.